bgpd/bgpd.c

```diff
@@ -278,17 +282,27 @@ void bgp_router_id_zebra_bump(vrf_id_t vrf_id, const struct prefix *router_id)
 			if (bgp->inst_type == BGP_INSTANCE_TYPE_VRF)
 				continue;
 
-			bgp->router_id_zebra = router_id->u.prefix4;
+			/* If router_id is NULL, then it is interface delete case
+			 * Set bgp router ID to the value router_id_zebra
```

Comment:
The 2nd comment I mentioned is to move the below lines of code to a common function as this is exactly same.

// ***************/
if(addr)
    bgp->router_id_zebra = *addr;
else
    addr = &bgp->router_id_zebra;
 
if (!bgp->router_id_static.s_addr)
	bgp_router_id_set(bgp, &router_id->u.prefix4);					

/*********************/

But thats fine. I think you can go ahead with the check in process